app/townsquare/views.py

```diff
@@ -0,0 +1,264 @@
+                offer = None
+        offers_by_category[key] = {
+            'offer': offer,
+            'time': next_time_available.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        }
+
+    # subscriber info
+    is_subscribed = False
+    if request.user.is_authenticated:
+        email_subscriber = request.user.profile.email_subscriptions.first()
+        if email_subscriber:
+            is_subscribed = email_subscriber.should_send_email_type_to('new_bounty_notifications')
+
+    # announcements
+    announcements = Announcement.objects.current()
+
+    # title
+    title = 'Home'
+    desc = 'View the recent activity on the Gitcoin network'
+    if "activity:" in tab:
+        try:
+            pk = int(tab.split(':')[1])
+            activity = Activity.objects.get(pk=pk)
+            title = f"@{activity.profile.handle}'s comment on Gitcoin "
+            desc = f"foo - {activity.text}"
+        except:
+            pass
+
+
+    # render page context
+    context = {
+        'title': title,
+        'card_desc': desc,
+        'nav': 'home',
+        'target': f'/activity?what={tab}',
+        'tab': tab,
+        'tabs': tabs,
+        'announcements': announcements,
+        'is_subscribed': is_subscribed,
+        'offers_by_category': offers_by_category,
+    }
+    return TemplateResponse(request, 'townsquare/index.html', context)
+
+
+@ratelimit(key='ip', rate='10/m', method=ratelimit.UNSAFE, block=True)
+@csrf_exempt
+def emailsettings(request):
+
+    if not request.user.is_authenticated:
+        raise Http404
+
+    is_subscribed = False
+    if request.user.is_authenticated:
+        email_subscriber = request.user.profile.email_subscriptions.first()
+        if email_subscriber:
+            is_subscribed = email_subscriber.should_send_email_type_to('new_bounty_notifications')
+
+            for key in request.POST.keys():
+                email_subscriber.set_should_send_email_type_to(key, bool(request.POST.get(key) == 'true'))
+                email_subscriber.save()
+
+    response = {}
+    return JsonResponse(response)
+
+
+@ratelimit(key='ip', rate='10/m', method=ratelimit.UNSAFE, block=True)
+@csrf_exempt
+def api(request, activity_id):
+
+    # pull back the obj
+    try:
+        activity = Activity.objects.get(pk=activity_id)
+    except:
+        raise Http404
+
+    # setup response
+    response = {}
+
+    # check for permissions
+    has_perms = request.user.is_authenticated
+    if request.POST.get('method') == 'delete':
+        has_perms = activity.profile == request.user.profile
+    if not has_perms:
+        raise Http404
+
+    # deletion request
+    if request.POST.get('method') == 'delete':
+        activity.delete()
+
+    # like request
+    elif request.POST.get('method') == 'like':
+        if request.POST['direction'] == 'liked':
+            Like.objects.create(profile=request.user.profile, activity=activity)
+        if request.POST['direction'] == 'unliked':
```

Comment:
Whats the difference between deleting a like or unliking?

Reply:
theres no difference